stats/statsview.cpp

```diff
@@ -716,54 +719,70 @@ void StatsView::QuartileMarker::updatePosition()
 		      x + quartileMarkerSize / 2.0, y);
 }
 
-StatsView::LineMarker::LineMarker(QPointF from, QPointF to, QPen pen, QtCharts::QChart *chart, StatsAxis *xAxis, StatsAxis *yAxis) :
+StatsView::RegressionLine::RegressionLine(double a, double b, QPen pen, QtCharts::QChart *chart, StatsAxis *xAxis, StatsAxis *yAxis) :
 	item(new QGraphicsLineItem(chart)),
 	xAxis(xAxis), yAxis(yAxis),
-	from(from), to(to)
+	a(a), b(b)
 {
 	item->setZValue(ZValues::chartFeatures);
 	item->setPen(pen);
-	updatePosition();
 }
 
-void StatsView::LineMarker::updatePosition()
+void StatsView::RegressionLine::updatePosition()
 {
 	if (!xAxis || !yAxis)
 		return;
-	double x1 = xAxis->toScreen(from.x());
-	double y1 = yAxis->toScreen(from.y());
-	double x2 = xAxis->toScreen(to.x());
-	double y2 = yAxis->toScreen(to.y());
-	item->setLine(x1, y1, x2, y2);
-}
-
-void StatsView::addLinearRegression(double a, double b, double minX, double maxX, double minY, double maxY, StatsAxis *xAxis, StatsAxis *yAxis)
-{
-	// Sanity check: line above or below chart
+	auto [minX, maxX] = xAxis->minMax();
+	auto [minY, maxY] = yAxis->minMax();
 	double y1 = a * minX + b;
 	double y2 = a * maxX + b;
-	if ((y1 <= minY && y2 <= minY) || (y1 >= maxY && y2 >= maxY))
-		return;
 
-	// If not fully inside drawing region, do clipping. With the check above this guarantees that a != 0,
-	// but owing to floating point imprecision, let's test again.
+	// If not fully inside drawing region, do clipping.
 	if ((y1 < minY || y1 > maxY || y2 < minY || y2 > maxY) && fabs(a) > 0.0001) {
 		// Intersections with y = minY and y = maxY lines
 		double intersect_x1 = (minY - b) / a;
 		double intersect_x2 = (maxY - b) / a;
-		if (intersect_x1 > intersect_x2)
+		if (intersect_x1 < intersect_x2)
```

Comment:
So you switched this comparison back... that was one of the things that I changed to fix the clipping.
With this branch I again get things like this
<img width="791" alt="Screen Shot 2021-01-06 at 10 31 47 AM" src="https://user-images.githubusercontent.com/237471/103806825-6d628400-500a-11eb-8daf-381a3cc4369e.png">

Reply:
merge error - corrected.